main.js

```diff
@@ -1,20 +1,46 @@
 const coffeeShop = {
   beans: 40,
 
+  beanPrice: 0.2,
+  
+  money: 5000,
+
   drinkRequirements: {
-    latte: 10,
-    americano: 5,
-    doubleShot: 15,
-    frenchPress: 12
+    latte: { beanRequirements: 10, price: 5 },
+    americano: { beanRequirements: 5, price: 3 },
+    doubleShot: { beanRequirements: 15, price: 8 },
+    frenchPress: { beanRequirements: 12, price: 7 }
   },
 
   makeDrink: function (drinkType) {
-    // TODO: Finish this method
+    const toMake = this.drinkRequirements[drinkType]
+    if (!toMake) {
+      console.log("Sorry, we don't make " + drinkType)
+      return false;
+    }
+    if (toMake.beanRequirements > this.beans) {
+      console.log("Sorry, we're all out of beans!")
+      return false;
+    }
```

Comment:
functions should do one thing
the purpose of `makeDrink` is to make a drink
you can extract the validations ( `if !toMake`, `if toMake.beanRequire...`) to a separate function, so your code will look like this:
```
makeDrink(drinkType){
  if(!validOrder(drinkType)){
    return
  }
  this.beans -= toMake.beanRequirements...
```
^in this case, our `makeDrink` function is truly focused on one thing only - making a drink